nemo/lightning/pytorch/accelerate/transformer_engine.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from types import MethodType
+
+import torch
+from transformer_engine import pytorch as te
+
+
+def accelerate(model, fp8_autocast=False):
```

Comment:
why not make the name explicit that this is for transformer engine? accelerate otherwise is very generic

Reply:
Good point, it was in a class before. Actually it's in a transformer_engine.py file. Let me think about this.

Reply:
+1 , WDYT about `apply_te_optimizations` ?

Reply:
Changed it to te_accelerate. I think it makes sense to use accelerate since that's the reason we use TE. But if you guys can think of a better function name, please suggest.

te_optimization or apply_te_optimization sound like an optimizer. So, that's why I don't think we should use the optimization keyword.